plugins/inputs/internet_speed/internet_speed.go

```diff
@@ -61,11 +66,11 @@ func (is *InternetSpeed) Gather(acc telegraf.Accumulator) error {
 	if err != nil {
 		return fmt.Errorf("ping test failed: %w", err)
 	}
-	err = is.server.DownloadTest(is.MemorySavingMode)
+	err = is.server.MultiDownloadTestContext(context.Background(), is.servers)
 	if err != nil {
 		return fmt.Errorf("download test failed, try `memory_saving_mode = true` if this fails consistently: %w", err)
 	}
-	err = is.server.UploadTest(is.MemorySavingMode)
+	err = is.server.MultiUploadTestContext(context.Background(), is.servers)
```

Comment:
Hi, @powerjs, Yes, it makes sense. As you are worried, most servers in Speedtest are provided by individuals or network suppliers, their quality and maximum bandwidth are also different. If we use a single server that has a small bandwidth server, the result will not be close to the ideal Internet speed. Even get a completely worthless value. In other words, when we have repeatedly measured the same server, the results obtained are very uncertain, because we have completely believed this only server at this time. This breaks away from the original purpose (obtaining a relatively ideal Internet Speed).

The difference between the main server and the auxiliary server is that it will always be selected and as a reference ping result. And the main server has the priority to connect.

In addition, I think there is a misunderstanding here🤔. The average speed referred to in is.server.DLSpeed and is.server.ULSpeed is to average the time window, rather than the average server count.

Thank you!

Reply:
> Single server option is still worth being retained. 

good, let's keep it and add an option called `server_mode = "single"` that defaults to the current behavior of a single server. It can also take `"multiple"` as an option, and will instead use this multiple tests.

> The difference between the main server and the auxiliary server is that it will always be selected and as a reference ping result. And the main server has the priority to connect.

However, this leads us to a question about the metrics this plugin generates. Currently we produce a metric like:

```s
internet_speed,host=speedtest.pocketinet.com.prod.hosts.ooklaserver.net:8080,server_id=39915 download=667.708142,upload=31.75404,latency=33.037773,jitter=1.640157 1678198233000000000
```

Do we need to update the resulting metric to clarify that they are using multiple servers rather than just the one listed above?

Reply:
@srebhan do you have thoughts on how to record the new behavior where we might be using multiple servers?

I've thought about the following when using multiple servers:

1) adding a new tag (e.g. `multiple`)
2) removing the host and server_id tags
3) create a new metric (e.g. `internet_speed_multi`)

Reply:
> In my understanding the multi-test produces the same metrics as the single one. Is this correct?  

Yes. 


The last question, should we mark the rate of each server when using multi mode?

Reply:
I think I would prefer we were consistent and only returned the total in both cases. Does that sound ok?

Reply:
Modifying the tag will break backward compatibility. I also think that we should keep it simple.